### ISSUE TITLE
ZWave: Thermostat mode 0x13 (direct control) not supported by the binding

### ISSUE THREAD

**Issue by spacemanspiff2007** [openhab/openhab#2819]
Hi
As described earlier (#2803) the direct control for thermostats is not supported by the binding.
Hi,

I have another Problem:
The Device supports the following Set-Modes:
````
    Frost Protection: BASIC SET 0x01, THERMOSTAT MODE SET 0x00
    Energy Saving: BASIC SET 0x00, THERMOSTAT MODE SET 0x0b
    Comfort: BASIC SET 0xff, THERMOSTAT MODE SET 0x01
    Direct Control: BASIC SET 0xfe, THERMOSTAT MODE SET 0x1f
````
I thought it would be clever to control the valve directly, that's why I bought the StallaZ.
However, the Thermostat-Mode doesn't seem to be applied to the Device.
````
    //Put in Manual state
    sendCommand( Schlafzimmer_StellaZ_Mode, 0x1F)
````

The message is not been sent:
````
2015-06-28 18:20:52.123 [DEBUG] [.z.internal.ZWaveActiveBinding:450 ]- NODE 20: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_MODE, value = 1
2015-06-28 18:20:52.124 [DEBUG] [.z.i.p.s.ZWaveCommandProcessor:63  ]- Sent message Message: class = SendData (0x13), type = Request (0x00), payload = 14 02 40 02 
2015-06-28 18:20:52.125 [DEBUG] [.z.i.p.s.ZWaveCommandProcessor:64  ]- Recv message Message: class = ApplicationCommandHandler (0x04), type = Request (0x00), payload = 00 14 03 40 03 01 
2015-06-28 18:20:52.125 [DEBUG] [.z.i.p.s.ZWaveCommandProcessor:65  ]- Checking transaction complete: class=ApplicationCommandHandler, expected=ApplicationCommandHandler, cancelled=false
2015-06-28 18:20:52.126 [DEBUG] [.z.i.p.s.ZWaveCommandProcessor:68  ]-          transaction complete!
2015-06-28 18:20:52.126 [DEBUG] [b.z.i.protocol.ZWaveController:682 ]- Notifying event listeners: ZWaveTransactionCompletedEvent
2015-06-28 18:20:52.127 [DEBUG] [.z.internal.ZWaveActiveBinding:433 ]- ZwaveIncomingEvent
2015-06-28 18:20:52.128 [DEBUG] [b.z.i.protocol.ZWaveController:985 ]- Callback ID = 76
2015-06-28 18:20:52.128 [DEBUG] [b.z.i.protocol.ZWaveController:667 ]- Enqueueing message. Queue length = 2
2015-06-28 18:20:52.129 [DEBUG] [WaveController$ZWaveSendThread:1374]- NODE 20: Response processed after 38ms/215ms.
2015-06-28 18:20:52.130 [DEBUG] [WaveController$ZWaveSendThread:1258]- Took message from queue for sending. Queue length = 1
2015-06-28 18:20:52.131 [DEBUG] [o.b.z.i.protocol.SerialMessage:233 ]- Assembled message buffer = 01 09 00 13 14 02 26 02 25 4B B9 
2015-06-28 18:20:52.133 [DEBUG] [WaveController$ZWaveSendThread:1315]- NODE 20: Sending REQUEST Message = 01 09 00 13 14 02 26 02 25 4B B9 
2015-06-28 18:20:52.140 [DEBUG] [eController$ZWaveReceiveThread:1528]- Receive Message = 01 04 01 13 01 E8 
2015-06-28 18:20:52.141 [DEBUG] [eController$ZWaveReceiveThread:1452]- Receive queue ADD: Length=1
2015-06-28 18:20:52.141 [DEBUG] [b.z.i.protocol.ZWaveController:1210]- Receive queue TAKE: Length=0
2015-06-28 18:20:52.142 [DEBUG] [o.b.z.i.protocol.SerialMessage:233 ]- Assembled message buffer = 01 04 01 13 01 E8 
2015-06-28 18:20:52.143 [DEBUG] [b.z.i.protocol.ZWaveController:1211]- Process Message = 01 04 01 13 01 E8 
2015-06-28 18:20:52.144 [DEBUG] [b.z.i.protocol.ZWaveController:190 ]- Message: class = SendData (0x13), type = Response (0x01), payload = 01 
2015-06-28 18:20:52.144 [DEBUG] [b.z.i.p.s.SendDataMessageClass:38  ]- NODE 20: Sent Data successfully placed on stack.
2015-06-28 18:20:52.152 [DEBUG] [eController$ZWaveReceiveThread:1528]- Receive Message = 01 05 00 13 4B 00 A2 
2015-06-28 18:20:52.154 [DEBUG] [eController$ZWaveReceiveThread:1452]- Receive queue ADD: Length=1
2015-06-28 18:20:52.154 [DEBUG] [c.ZWaveThermostatModeConverter:97  ]- NODE 20: receiveCommand with converter class org.openhab.binding.zwave.internal.converter.command.IntegerCommandConverter 
2015-06-28 18:20:52.154 [DEBUG] [b.z.i.protocol.ZWaveController:1210]- Receive queue TAKE: Length=0
2015-06-28 18:20:52.155 [DEBUG] [WaveThermostatModeCommandClass:256 ]- NODE 20: setValueMessage 31, modeType empty false
2015-06-28 18:20:52.156 [DEBUG] [o.b.z.i.protocol.SerialMessage:233 ]- Assembled message buffer = 01 07 00 13 4B 00 00 00 A0 
2015-06-28 18:20:52.157 [ERROR] [WaveThermostatModeCommandClass:265 ]- NODE 20: Unsupported mode type 31
2015-06-28 18:20:52.157 [DEBUG] [b.z.i.protocol.ZWaveController:1211]- Process Message = 01 07 00 13 4B 00 00 00 A0 
2015-06-28 18:20:52.158 [DEBUG] [c.ZWaveThermostatModeConverter:100 ]- NODE 20: receiveCommand sending message null 
2015-06-28 18:20:52.159 [DEBUG] [b.z.i.protocol.ZWaveController:190 ]- Message: class = SendData (0x13), type = Request (0x00), payload = 4B 00 
2015-06-28 18:20:52.159 [WARN ] [c.ZWaveThermostatModeConverter:102 ]- NODE 20: Generating message failed for command class = THERMOSTAT_MODE, endpoint = 0
````

**Comment by cdjackson** [comment]
Just to be 100% sure, you’re using the latest off cloudbees today?  If so, can I ask you to try again tomorrow before I look into this since I believe a change to this issue was merged on Friday - see here (https://github.com/openhab/openhab/pull/2808 <https://github.com/openhab/openhab/pull/2808>). I would have thought it would be in todays snapshot, but it might just be that it’s not made it through the cloudbees build system yet...

**Comment by spacemanspiff2007** [comment]
Hi,
I downloaded it after you recommended it.
File-Date shows 27.06.2015 - 03:21 which corresponds to the change history.
Is there the build-number or something hidden somewhere?
Otherwise I'll download it again tomorrow ... .

**Comment by cdjackson** [comment]
> On 28 Jun 2015, at 18:32, spacemanspiff2007 <notifications@github.com> wrote:
> 
> Is there the build-number or something hidden somewhere?

Yes - if you look in the zwave log, search for "Z-Wave binding started” - there’s a version/build date on this entry.

**Comment by spacemanspiff2007** [comment]
````2015-06-28 18:10:17.404 [DEBUG] [.zwave.internal.ZWaveActivator:36  ]- Z-Wave binding started. Version 1.8.0.201506270113
````

**Comment by cdjackson** [comment]
I just checked the change log on cloudbees and it looks like this should have been included. I’m not sure I’ll get to look at this tonight...

**Comment by cdjackson** [comment]
Firstly, can you post the XML file for this node (from /etc/zwave).
Then, either delete the XML file and restart the binding… OR - if you have HABmin, you should be able to click on the node, and click the “reinitialise node” button (it might have a slightly different name).  This might solve the problem.
Then repost the XML...

**Comment by spacemanspiff2007** [comment]
Original XML:
````
<node>
  <deviceClass>
    <basicDeviceClass>ROUTING_SLAVE</basicDeviceClass>
    <genericDeviceClass>THERMOSTAT</genericDeviceClass>
    <specificDeviceClass>THERMOSTAT_GENERAL_V2</specificDeviceClass>
  </deviceClass>
  <homeId>0x184d8a6</homeId>
  <nodeId>20</nodeId>
  <version>4</version>
  <name>Heizung</name>
  <location>Schlafzimmer</location>
  <manufacturer>0x148</manufacturer>
  <deviceId>0x1</deviceId>
  <deviceType>0x1</deviceType>
  <listening>false</listening>
  <frequentlyListening>false</frequentlyListening>
  <routing>true</routing>
  <supportedCommandClasses>
    <entry>
      <commandClass>WAKE_UP</commandClass>
      <WakeUpCommandClass>
        <version>2</version>
        <instances>1</instances>
        <targetNodeId>1</targetNodeId>
        <interval>240</interval>
        <minInterval>240</minInterval>
        <maxInterval>15728400</maxInterval>
        <defaultInterval>604672</defaultInterval>
        <intervalStep>240</intervalStep>
        <lastWakeup>2015-06-28 16:20:51.785 UTC</lastWakeup>
      </WakeUpCommandClass>
    </entry>
    <entry>
      <commandClass>NO_OPERATION</commandClass>
      <noOperationCommandClass>
        <version>1</version>
        <instances>1</instances>
      </noOperationCommandClass>
    </entry>
    <entry>
      <commandClass>THERMOSTAT_MODE</commandClass>
      <thermostatModeCommandClass>
        <version>2</version>
        <instances>1</instances>
        <modeTypes>
          <modeType>OFF</modeType>
          <modeType>HEAT</modeType>
          <modeType>HEAT_ECON</modeType>
        </modeTypes>
        <isGetSupported>true</isGetSupported>
      </thermostatModeCommandClass>
    </entry>
    <entry>
      <commandClass>SWITCH_MULTILEVEL</commandClass>
      <multiLevelSwitchCommandClass>
        <version>3</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </multiLevelSwitchCommandClass>
    </entry>
    <entry>
      <commandClass>BASIC</commandClass>
      <basicCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </basicCommandClass>
    </entry>
    <entry>
      <commandClass>SENSOR_MULTILEVEL</commandClass>
      <multiLevelSensorCommandClass>
        <version>4</version>
        <instances>1</instances>
        <sensors>
          <entry>
            <multilevelSensorType>TEMPERATURE</multilevelSensorType>
            <multilevelSensor>
              <sensorType>TEMPERATURE</sensorType>
              <initialised>true</initialised>
            </multilevelSensor>
          </entry>
        </sensors>
        <isGetSupported>true</isGetSupported>
      </multiLevelSensorCommandClass>
    </entry>
    <entry>
      <commandClass>THERMOSTAT_SETPOINT</commandClass>
      <thermostatSetpointCommandClass>
        <version>2</version>
        <instances>1</instances>
        <setpoints>
          <entry>
            <setpointType>FURNACE</setpointType>
            <org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
              <setpointType>FURNACE</setpointType>
              <initialised>true</initialised>
              <initCount>4</initCount>
              <outer-class reference="../../../.."/>
            </org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
          </entry>
          <entry>
            <setpointType>HEATING</setpointType>
            <org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
              <setpointType>HEATING</setpointType>
              <initialised>true</initialised>
              <initCount>0</initCount>
              <outer-class reference="../../../.."/>
            </org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
          </entry>
        </setpoints>
        <isGetSupported>true</isGetSupported>
      </thermostatSetpointCommandClass>
    </entry>
    <entry>
      <commandClass>VERSION</commandClass>
      <versionCommandClass>
        <version>1</version>
        <instances>1</instances>
        <libraryType>LIB_SLAVE_ENHANCED</libraryType>
        <protocolVersion>3.42</protocolVersion>
        <applicationVersion>0.4</applicationVersion>
      </versionCommandClass>
    </entry>
    <entry>
      <commandClass>MANUFACTURER_SPECIFIC</commandClass>
      <manufacturerSpecificCommandClass>
        <version>1</version>
        <instances>1</instances>
      </manufacturerSpecificCommandClass>
    </entry>
    <entry>
      <commandClass>BATTERY</commandClass>
      <batteryCommandClass>
        <version>1</version>
        <instances>1</instances>
        <batteryLevel>10</batteryLevel>
        <batteryLow>false</batteryLow>
        <isGetSupported>true</isGetSupported>
      </batteryCommandClass>
    </entry>
  </supportedCommandClasses>
  <nodeNeighbors>
    <int>1</int>
    <int>6</int>
  </nodeNeighbors>
  <lastSent>2015-06-28 16:20:51.976 UTC</lastSent>
  <lastReceived>2015-06-28 16:20:52.12 UTC</lastReceived>
  <applicationUpdateReceived>true</applicationUpdateReceived>
</node>
````

**Comment by cdjackson** [comment]
> On 28 Jun 2015, at 19:08, spacemanspiff2007 <notifications@github.com> wrote:
> 
> Original XML:
Thanks - I think when you restart after deleting the XML, it will work...

**Comment by spacemanspiff2007** [comment]
Neu xml:
````
<node>
  <deviceClass>
    <basicDeviceClass>ROUTING_SLAVE</basicDeviceClass>
    <genericDeviceClass>THERMOSTAT</genericDeviceClass>
    <specificDeviceClass>THERMOSTAT_GENERAL_V2</specificDeviceClass>
  </deviceClass>
  <homeId>0x184d8a6</homeId>
  <nodeId>20</nodeId>
  <version>4</version>
  <name>Heizung</name>
  <location>Schlafzimmer</location>
  <manufacturer>0x148</manufacturer>
  <deviceId>0x1</deviceId>
  <deviceType>0x1</deviceType>
  <listening>false</listening>
  <frequentlyListening>false</frequentlyListening>
  <routing>true</routing>
  <supportedCommandClasses>
    <entry>
      <commandClass>VERSION</commandClass>
      <versionCommandClass>
        <version>1</version>
        <instances>1</instances>
        <libraryType>LIB_SLAVE_ENHANCED</libraryType>
        <protocolVersion>3.42</protocolVersion>
        <applicationVersion>0.4</applicationVersion>
      </versionCommandClass>
    </entry>
    <entry>
      <commandClass>WAKE_UP</commandClass>
      <WakeUpCommandClass>
        <version>2</version>
        <instances>1</instances>
        <targetNodeId>1</targetNodeId>
        <interval>240</interval>
        <minInterval>240</minInterval>
        <maxInterval>15728400</maxInterval>
        <defaultInterval>604672</defaultInterval>
        <intervalStep>240</intervalStep>
        <lastWakeup>2015-06-28 18:17:47.955 UTC</lastWakeup>
      </WakeUpCommandClass>
    </entry>
    <entry>
      <commandClass>SENSOR_MULTILEVEL</commandClass>
      <multiLevelSensorCommandClass>
        <version>4</version>
        <instances>1</instances>
        <sensors>
          <entry>
            <multilevelSensorType>TEMPERATURE</multilevelSensorType>
            <multilevelSensor>
              <sensorType>TEMPERATURE</sensorType>
              <initialised>true</initialised>
            </multilevelSensor>
          </entry>
        </sensors>
        <isGetSupported>true</isGetSupported>
      </multiLevelSensorCommandClass>
    </entry>
    <entry>
      <commandClass>THERMOSTAT_SETPOINT</commandClass>
      <thermostatSetpointCommandClass>
        <version>2</version>
        <instances>1</instances>
        <setpoints>
          <entry>
            <setpointType>HEATING</setpointType>
            <org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
              <setpointType>HEATING</setpointType>
              <initialised>true</initialised>
              <initCount>0</initCount>
              <outer-class reference="../../../.."/>
            </org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
          </entry>
          <entry>
            <setpointType>FURNACE</setpointType>
            <org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
              <setpointType>FURNACE</setpointType>
              <initialised>true</initialised>
              <initCount>0</initCount>
              <outer-class reference="../../../.."/>
            </org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveThermostatSetpointCommandClass_-Setpoint>
          </entry>
        </setpoints>
        <isGetSupported>true</isGetSupported>
      </thermostatSetpointCommandClass>
    </entry>
    <entry>
      <commandClass>MANUFACTURER_SPECIFIC</commandClass>
      <manufacturerSpecificCommandClass>
        <version>1</version>
        <instances>1</instances>
      </manufacturerSpecificCommandClass>
    </entry>
    <entry>
      <commandClass>NO_OPERATION</commandClass>
      <noOperationCommandClass>
        <version>1</version>
        <instances>1</instances>
      </noOperationCommandClass>
    </entry>
    <entry>
      <commandClass>BATTERY</commandClass>
      <batteryCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </batteryCommandClass>
    </entry>
    <entry>
      <commandClass>THERMOSTAT_MODE</commandClass>
      <thermostatModeCommandClass>
        <version>2</version>
        <instances>1</instances>
        <modeTypes>
          <modeType>OFF</modeType>
          <modeType>MANUAL</modeType>
          <modeType>HEAT</modeType>
          <modeType>HEAT_ECON</modeType>
        </modeTypes>
        <isGetSupported>true</isGetSupported>
      </thermostatModeCommandClass>
    </entry>
    <entry>
      <commandClass>BASIC</commandClass>
      <basicCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </basicCommandClass>
    </entry>
    <entry>
      <commandClass>SWITCH_MULTILEVEL</commandClass>
      <multiLevelSwitchCommandClass>
        <version>3</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </multiLevelSwitchCommandClass>
    </entry>
  </supportedCommandClasses>
  <nodeNeighbors>
    <int>1</int>
    <int>6</int>
  </nodeNeighbors>
  <lastSent>2015-06-28 18:17:47.825 UTC</lastSent>
  <lastReceived>2015-06-28 18:17:46.813 UTC</lastReceived>
  <applicationUpdateReceived>true</applicationUpdateReceived>
</node>
````

**Comment by cdjackson** [comment]
> On 28 Jun 2015, at 19:21, spacemanspiff2007 <notifications@github.com> wrote:
> 
>         <modeTypes>
>           <modeType>OFF</modeType>
>           <modeType>MANUAL</modeType>
>           <modeType>HEAT</modeType>
>           <modeType>HEAT_ECON</modeType>
>         </modeTypes>
Does it work now? I see it’s now reporting that it supports MANUAL mode, which it didn’t before...

**Comment by spacemanspiff2007** [comment]
I am still figuring out the Thermostat Setpoint - mode.
Because it didn't work I changed my rules to the THERMOSTAT_SETPOINT, but because of the infrequent wakeup of the device testing is very hard.
It seems that the wakeup period is not long enough so that all messages get delivered, but I am not sure yet.
After some more testing I'll switch back to manual mode and let you know.
Thank you for your help!

**Comment by andreground** [comment]
This has been fixed by my two recent commits to ZWaveThermostatModeCommandClass.java.
I currently have them working fine in Direct Control mode. Wakeup period of these devices is very short so messages may be kept in queue for long periods, thus my suggestion is to set Wakeup interval to the lowest possible value (2400 seconds).

**Comment by spacemanspiff2007** [comment]
Hi,
I will try to test this on Saturday or Sunday. On my devices it is possible to set the Wakeup-Interval to 240s, but the wakeup time is very short so sometimes not all messages go through.

**Comment by cdjackson** [comment]
Can we close this?

**Comment by spacemanspiff2007** [comment]
I am sorry that I only made promises and didn't contribute as I told you.
Some private stuff popped up and I was on holiday.
I sheduled the test for the weekend - I haven't forgotten.
Sorry.

**Comment by cdjackson** [comment]
No probs - thanks.

**Comment by spacemanspiff2007** [comment]
Hi,
Message gets created:
````
2015-08-19 20:10:58.009 [DEBUG] [c.ZWaveThermostatModeConverter:97  ]- NODE 20: receiveCommand with converter class org.openhab.binding.zwave.internal.converter.command.IntegerCommandConverter 
2015-08-19 20:10:58.011 [DEBUG] [WaveThermostatModeCommandClass:256 ]- NODE 20: setValueMessage 31, modeType empty false
2015-08-19 20:10:58.013 [DEBUG] [WaveThermostatModeCommandClass:269 ]- NODE 20: Creating new message for application command THERMOSTAT_MODE_SET
2015-08-19 20:10:58.014 [DEBUG] [o.b.z.i.protocol.SerialMessage:109 ]- NODE 20: Creating empty message of class = SendData (0x13), type = Request (0x00)
2015-08-19 20:10:58.016 [DEBUG] [c.ZWaveThermostatModeConverter:100 ]- NODE 20: receiveCommand sending message Message: class = SendData (0x13), type = Request (0x00), payload = 14 03 40 01 1F  
````

Wake up with old status:
````
2015-08-19 20:11:47.249 [DEBUG] [.z.internal.ZWaveActiveBinding:450 ]- NODE 20: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_MODE, value = 1
````
Next Wake up:
````
2015-08-19 20:23:51.153 [DEBUG] [.z.internal.ZWaveActiveBinding:450 ]- NODE 20: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_MODE, value = 31
````

Switching modes works! :-)

**Comment by cdjackson** [comment]
Thanks :)

**Comment by Vitchling** [comment]
Hi spacemanspiff
I've been running OH 1.5.1 for the last year or so, with 4 x StellaZs and also wanted to use direct control but it wasn't supported until you raised this. I found a very clunky way to get around it but I always wanted a better solution.

I've now finally got OH 1.7.0 working and would love to go to direct valve control. I'm not a programmer though, so most of my efforts take forever.

Would you be so kind as to share your StellaZ items configs and maybe a rule or two for setting valve levels?

I use a separate temperature sensor in each room as I found the StellaZ thermometer was more affected by the hot water pipe temp than room temp so I guess I'll be ok with most of my control rules, it'll just be how to set the mode to manual, then how to set the valve level.

Thank you
Vitchling

**Comment by spacemanspiff2007** [comment]
Hi,

of course I'll share my config.
Be aware that if you poll a lot of values, I found the wakeup and response of the device rather random and unreliable.
This is why I (after lots of anger) finally switched to normal application mode and set the temperature as desired.

This is what I got out of my repository:
```
Number	Schlafzimmer_Heizung_Battery			"Batterie [%d %%]"									(hz)							{ zwave="20:command=BATTERY" }
Number	Schlafzimmer_Heizung_Mode				"Mode [%d]"											(hz)							{ zwave="20:command=THERMOSTAT_MODE,refresh_interval=240" }
Number	Schlafzimmer_Heizung_Temperatur_Ist		"Heizung Ist-Temperatur [%.1f]"						(hz,cTemperatur,gRRD4J)			{ zwave="20:command=SENSOR_MULTILEVEL,sensor_type=1,refresh_interval=240" }
Number	Schlafzimmer_Heizung_Temperatur_Soll	"Heizung SollTemperatur [%.1f]"						(hz,gRRD4J)						{ zwave="20:command=THERMOSTAT_SETPOINT,setpoint_scale=0,refresh_interval=240" }
Number	Schlafzimmer_Heizung_Pos_Ist			"Ventil Pos Ist [%d]"								(hz,gRRD4J,gInitializeZero)		{ zwave="20:command=SWITCH_MULTILEVEL,sensor_type=1,refresh_interval=240" }
```



```
rule "StellaZ Set Manual Mode"
when
	Item Schlafzimmer_StellaZ_Mode changed
then
	
	var Number state =  Schlafzimmer_StellaZ_Mode.state as DecimalType
	logDebug( "Schlafzimmer", "Thermostat mode: " + state)
	
	//Basic somehow doesn't work!
	//var Number state_basic =  Schlafzimmer_StellaZ_Mode_Basic.state as DecimalType
	//logDebug( "Schlafzimmer", "Basic mode: " + state_basic)
	
	//Already desired state
	if( state == 0x1F) return true

	//Put in Manual state
	sendCommand( Schlafzimmer_StellaZ_Mode, 0x1F)
end


//Fenster auf
rule "StellaZ Fenster auf"
when
	Item Schlafzimmer_MS1_Fenster changed from CLOSED to OPEN
then
	logDebug( "Schlafzimmer", "Fenster offen!")
	postUpdate( Schlafzimmer_StellaZ_Pos_Soll, 0)
end

//Force 
rule "StellaZ Set Valve Position"
when
	Item Schlafzimmer_StellaZ_Pos_Ist changed or
	Item Schlafzimmer_StellaZ_Pos_Soll changed
then
	var Number ist	= Schlafzimmer_StellaZ_Pos_Ist.state	as DecimalType
	var Number soll	= Schlafzimmer_StellaZ_Pos_Soll.state	as DecimalType

	logDebug( "Schlafzimmer", "Ventilposition Ist: " + ist + " Soll: " + soll)
	if( ist == soll) return false
	
	//Erneut senden
	sendCommand( Schlafzimmer_StellaZ_Pos_Ist, soll)
end
```

**Comment by Vitchling** [comment]
Hi
 
Thanks for sharing your config.
 
It sounds like it didn't work reliably and you reverted to setting a target
temperature instead (which is what I do now).
 
That's disappointing as I hoped you had a cure for some of my own problems
with this. Hmph.
 
I have a test radiator with a StellaZ in a room I don't use, so I will try
out your ideas and see if I can learn from them.
 
For information, I found the StellarZ thermostat to be wildly unreliable
compared to room temperatures. So, I set the StellarZ target temperatures to
28c when I want them to come on and 10c when I want them to switch off and
control that with a PIR/TEMP sensor in each room. That seems to keep the
rooms within about 0.5c of the actual target temperature but it's also
integrated with my central heating demand controller and other radiators,
some of which don't have StellaZ.
 
Downstairs (on a zwave boiler control) I have Stellarz's on my living room
and dining room/office. My hall and both kitchen radiators are on standard
TRVs. I can pretty much fine control the temperature of all these areas,
within limits.
 
Upstairs I still use a standard thermostat in my bedroom but have a StellarZ
in the bathroom and can control my bedroom and bathroom temperatures quite
finely too.
 
It kind of works well and I control each room using a Gcal calander
schedule, which is a great way to set your heating up if you have a variying
schedule. It's not perfect though. It screws up every time I have to update
openHAB or re-authorise Gcal for each new openHAB release. 
 
I'm begining to think that there's a balance between effort and benefit.
Some of the older methods were a bit clunky and inefficient but they always
worked...
 
Anyway, thanks again for your info. I'll use it with interest.
 
Cheers
Vitchling
 
 
 
 


  _____
